rotkehlchen/db/upgrades/v32_v33.py

```diff
@@ -71,6 +74,376 @@ def _create_new_tables(cursor: 'DBCursor') -> None:
 """)
 
 
+def _force_bytes_for_tx_hashes(cursor: 'DBCursor') -> None:
+    # aave_events
+    cursor.execute("""
+    CREATE TABLE aave_events_copy (
+        address VARCHAR[42] NOT NULL,
+        event_type VARCHAR[10] NOT NULL,
+        block_number INTEGER NOT NULL,
+        timestamp INTEGER NOT NULL,
+        tx_hash BLOB NOT NULL,
+        log_index INTEGER NOT NULL,
+        asset1 TEXT NOT NULL,
+        asset1_amount TEXT NOT NULL,
+        asset1_usd_value TEXT NOT NULL,
+        asset2 TEXT,
+        asset2amount_borrowrate_feeamount TEXT,
+        asset2usd_value_accruedinterest_feeusdvalue TEXT,
+        borrow_rate_mode VARCHAR[10],
+        FOREIGN KEY(asset1) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        FOREIGN KEY(asset2) REFERENCES assets(identifier) ON UPDATE CASCADE,
+        PRIMARY KEY (event_type, tx_hash, log_index)
+    );
+    """)
+    aave_events = cursor.execute('SELECT * FROM aave_events')
```

Comment:
for 2. there's no built-in way to take a string and convert it to a blob

for 1. you mean copying everything verbatim from the previous table, then selecting all tx_hash columns and updating it? If that's the case, the number of db calls aren't reduced in any way

Reply:
Yeah I would also like to do 2. But there seems to be no built-in hex to bytes function in sqlite with a quick search as @prettyirrelevant wrote.

for (1) this is even more time consuming. Since you would need to copy `N` columns by inserting them into the new table and then go through `N` values and modify them again once `tx_hash` is modified in `python`. So it's `2N` the number of writes.